models/org_team.go

```diff
@@ -220,6 +224,25 @@ func (t *Team) addRepository(e Engine, repo *Repository) (err error) {
 	return nil
 }
 
+// addAllRepositories adds all repositories to the team.
+// If the team already has some repositories they will be left unchanged.
+func (t *Team) addAllRepositories(e Engine) error {
+	var orgRepos []Repository
+	if err := e.Where("owner_id = ?", t.OrgID).Find(&orgRepos); err != nil {
+		return fmt.Errorf("get org repos: %v", err)
+	}
+
+	for _, repo := range orgRepos {
```

Comment:
How about load all team repository ids and calc the need insert repository ids and insert them batch?

Reply:
I can try to get all repos to add in one sql query? This should get all repositories the team currently doesn't have (not tried it with xorm yet):
`
SELECT * FROM repository LEFT JOIN team_repo ON team_repo.org_id = repository.owner_id AND team_repo.repo_id = repository.id AND team_repo.team_id=<id> WHERE repository.owner_id=<org_id> AND team_repo.id IS NULL
`

Otherwise I am not sure it gains much or how to do it, there is quite much calculation in each add repository.

Reply:
OK. I think your solution is better than before.